lib/src/impl_ffi/impl_ffi.digest.dart

```diff
@@ -14,18 +14,15 @@
 
 part of 'impl_ffi.dart';
 
-abstract class _Hash implements Hash {
-  const _Hash();
+abstract class _HashImpl implements HashImpl {
+  const _HashImpl();
 
-  factory _Hash.fromHash(Hash hash) {
-    if (hash is _Hash) {
+  factory _HashImpl.fromHash(HashImpl hash) {
+    if (hash is _HashImpl) {
       return hash;
     }
-    throw ArgumentError.value(
-      hash,
-      'hash',
-      'Custom implementations of Hash is not supported',
-    );
+    throw AssertionError(
+        'Internal invariant violated: Custom implementations of HashImpl are not supported.');
```

Comment:
```suggestion
        'Custom implementations of HashImpl are not supported.');
```
An `AssertionError` is always an internal invariant that's violated.

Note. This is the same error we have in `_getHashAlgorithm` in `impl_js`.

Reply:
Noted 🫡